### ISSUE TITLE
error winbox ubuntu 24.04

### ISSUE THREAD

**Issue by fibulaotserv** [panaceya/winbox#52]
**Describe the bug**

**To Reproduce**

**Expected behavior**

**Screenshots**

**Distribution (please complete the following information):**
 - Name: [e.g. ubuntu, neon, debian]
 - Version: [e.g. 20.04, 18.04]
 - Architecture: [e.g. amd64 or i386 or ...]

** `snap version` **
```

```

** `snap info winbox` **
```
```

** `snap list  winbox` **
```
```

**Additional context**

**Comment by fibulaotserv** [comment]
error ubuntu 24.04

**Comment by mfth12** [comment]
icam't start winbox (snap) on ubuntu 24.04 too
![image](https://github.com/panaceya/winbox/assets/53788525/cf32262f-0fa2-452f-acf2-911021945784)

**Comment by cspencer51** [comment]
Having this same issue.

**Describe the bug**
The application fails to start.
```
$ winbox 
000b:err:module:__wine_process_init failed to load L"C:\\windows\\system32\\wineboot.exe", error c0000018
0009:err:environ:run_wineboot failed to start wineboot c0000018
 $ 000d:err:winediag:nodrv_CreateWindow Application tried to create a window, but no driver could be loaded.
000d:err:winediag:nodrv_CreateWindow The explorer process failed to start.
```

I also see these messages from AppArmor. I don't see these on Ubuntu 22.04:
```
[73778.766915] audit: type=1400 audit(1715194447.032:710): apparmor="DENIED" operation="open" class="file" profile="snap.winbox.winbox" name="/run/" pid=134765 comm="wineserver" requested_mask="r" denied_mask="r" fsuid=1002 ouid=0
[73778.853985] audit: type=1400 audit(1715194447.119:712): apparmor="DENIED" operation="connect" class="file" profile="snap.winbox.winbox" name="/run/cups/cups.sock" pid=134772 comm="winbox64.exe" requested_mask="wr" denied_mask="wr" fsuid=1002 ouid=0
```

**To Reproduce**
Steps to reproduce the behavior:
1. Run `winbox` from the command line _or_ launch the Winbox snap application.

**Expected behavior**
The application should start.

**Screenshots**
N/A

**Distribution (please complete the following information):**

Name: ubuntu
Version: 24.04
Architecture: amd64

** snap version **
```
snap    2.62+24.04build1
snapd   2.62+24.04build1
series  16
ubuntu  24.04
kernel  6.8.0-31-generic
```
** snap info winbox **
```
name:      winbox
summary:   small utility that allows administration of MikroTik RouterOS
publisher: ThePanaceya
store-url: https://snapcraft.io/winbox
license:   unset
description: |
  Winbox is a small utility that allows administration of MikroTik RouterOS using a fast and simple
  GUI.
  
  Snap conatins Wine runtime and winbox client.
  
  Readme: https://github.com/panaceya/winbox/blob/main/README.md
commands:
  - winbox
snap-id:      UdK6K4ZjRfjL0wRFh0tWGO7vPhA4Qge4
tracking:     latest/stable
refresh-date: yesterday at 15:12 PDT
channels:
  latest/stable:    3.40 2024-03-09 (135) 47MB -
  latest/candidate: ↑                          
  latest/beta:      ↑                          
  latest/edge:      3.40 2024-03-05 (135) 47MB -
installed:          3.40            (135) 47MB -
```
** snap list  winbox **
```
Name    Version  Rev  Tracking       Publisher    Notes
winbox  3.40     135  latest/stable  thepanaceya  -
```

**Comment by manumaiden** [comment]
I have the same issue:
```bash
$ winbox 
000b:err:module:__wine_process_init failed to load L"C:\\windows\\system32\\wineboot.exe", error c0000018
0009:err:environ:run_wineboot failed to start wineboot c0000018
$ 000d:err:winediag:nodrv_CreateWindow Application tried to create a window, but no driver could be loaded.
000d:err:winediag:nodrv_CreateWindow The explorer process failed to start.

$  journalctl --no-pager --follow 
May 16 18:31:33 X1Carbon systemd[3512]: Started snap.winbox.winbox-ef3415a8-2cb2-4745-b268-f0e0a35e753b.scope.
May 16 18:31:34 X1Carbon kernel: audit: type=1326 audit(1715877094.630:417): auid=1000 uid=1000 gid=1000 ses=62 subj=snap.winbox.winbox pid=22889 comm="wineserver" exe="/snap/winbox/135/wine-platform/wine-stable/bin/wineserver" sig=0 arch=c000003e syscall=203 compat=0 ip=0x7bff9a622f09 code=0x50000
May 16 18:31:35 X1Carbon kernel: audit: type=1400 audit(1715877095.123:418): apparmor="DENIED" operation="open" class="file" profile="snap.winbox.winbox" name="/run/" pid=22889 comm="wineserver" requested_mask="r" denied_mask="r" fsuid=1000 ouid=0
May 16 18:31:35 X1Carbon kernel: audit: type=1326 audit(1715877095.130:419): auid=1000 uid=1000 gid=1000 ses=62 subj=snap.winbox.winbox pid=22889 comm="wineserver" exe="/snap/winbox/135/wine-platform/wine-stable/bin/wineserver" sig=0 arch=c000003e syscall=203 compat=0 ip=0x7bff9a622f09 code=0x50000
May 16 18:31:35 X1Carbon kernel: audit: type=1400 audit(1715877095.342:420): apparmor="DENIED" operation="connect" class="file" profile="snap.winbox.winbox" name="/run/cups/cups.sock" pid=22893 comm="winbox64.exe" requested_mask="wr" denied_mask="wr" fsuid=1000 ouid=0
May 16 18:31:35 X1Carbon kernel: audit: type=1326 audit(1715877095.347:421): auid=1000 uid=1000 gid=1000 ses=62 subj=snap.winbox.winbox pid=22889 comm="wineserver" exe="/snap/winbox/135/wine-platform/wine-stable/bin/wineserver" sig=0 arch=c000003e syscall=101 compat=0 ip=0x7bff9a63661f code=0x50000
May 16 18:31:35 X1Carbon kernel: audit: type=1326 audit(1715877095.347:422): auid=1000 uid=1000 gid=1000 ses=62 subj=snap.winbox.winbox pid=22889 comm="wineserver" exe="/snap/winbox/135/wine-platform/wine-stable/bin/wineserver" sig=0 arch=c000003e syscall=101 compat=0 ip=0x7bff9a63661f code=0x50000
May 16 18:31:38 X1Carbon systemd[3512]: snap.winbox.winbox-ef3415a8-2cb2-4745-b268-f0e0a35e753b.scope: Consumed 1.631s CPU time.
```

systeminfo:

```bash
$ uname -a && cat /etc/os-release | head -2
Linux X1Carbon 6.8.0-31-generic #31-Ubuntu SMP PREEMPT_DYNAMIC Sat Apr 20 00:40:06 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux
PRETTY_NAME="Ubuntu 24.04 LTS"
NAME="Ubuntu"

$ snap info winbox
name:      winbox
summary:   small utility that allows administration of MikroTik RouterOS
publisher: ThePanaceya
store-url: https://snapcraft.io/winbox
license:   unset
description: |
  Winbox is a small utility that allows administration of MikroTik RouterOS using a fast and simple
  GUI.
  
  Snap conatins Wine runtime and winbox client.
  
  Readme: https://github.com/panaceya/winbox/blob/main/README.md
commands:
  - winbox
snap-id:      UdK6K4ZjRfjL0wRFh0tWGO7vPhA4Qge4
tracking:     latest/stable
refresh-date: today at 18:04 CEST
channels:
  latest/stable:    3.40 2024-03-09 (135) 47MB -
  latest/candidate: ↑                          
  latest/beta:      ↑                          
  latest/edge:      3.40 2024-03-05 (135) 47MB -
installed:          3.40            (135) 47MB -

$ snap version
snap    2.62+24.04build1
snapd   2.62+24.04build1
series  16
ubuntu  24.04
kernel  6.8.0-31-generic
```

Thank you

**Comment by gil-obradors** [comment]
Hello,
I have found that if you install winetricks it works.
Booting with wine says other errors, but surely editing the snap and adding to install also winetricks will solve it.


I will try it later



#47 #50  #51  ( are the same, can merge)

**Comment by manumaiden** [comment]
@gil-obradors not sure if I did correctly but it still doesn't work.
```bash
$ apt list | grep winetr
winetricks/noble,now 20240105-2 all [installed]
```
The `winbox` installed thru snap still fails to start. As workaround, I downloaded the winbox.exe from Mikrotik official website and I run with wine:
```bash
manumaiden@X1Carbon:~$ wine ./Downloads/winbox.exe 
wine: created the configuration directory '/home/manumaiden/.wine'
004c:err:ole:StdMarshalImpl_MarshalInterface Failed to create ifstub, hr 0x80004002
004c:err:ole:CoMarshalInterface Failed to marshal the interface {6d5140c1-7436-11ce-8034-00aa006009fa}, hr 0x80004002
004c:err:ole:apartment_get_local_server_stream Failed: 0x80004002
004c:err:ole:start_rpcss Failed to open RpcSs service
0054:err:ole:StdMarshalImpl_MarshalInterface Failed to create ifstub, hr 0x80004002
0054:err:ole:CoMarshalInterface Failed to marshal the interface {6d5140c1-7436-11ce-8034-00aa006009fa}, hr 0x80004002
0054:err:ole:apartment_get_local_server_stream Failed: 0x80004002
wine: configuration in L"/home/manumaiden/.wine" has been updated.
creating Window Class routeros_null
creating Window Class routeros_connect
DPI=96
EMS=13
ERROR: bad bmp format id=2329
biPlanes=1, biBitCount=1, biCompression=0
ERROR: bad bmp format id=3329
biPlanes=1, biBitCount=1, biCompression=0
creating Window Class routeros_dbl_canvas
rescaleDPI starting-dpi=96, zoom=0 result=96
 discovery started
rescaning
connecting to 192.168.88.255
connecting to 192.168.122.255
loadConfig C:\users\manumaiden\AppData\Roaming\Mikrotik\Winbox\Addresses.cdb
rescaning
```

There are few errors but it works :) .

**Comment by panaceya** [comment]
Fixed in #53. 
You need before refresh by `snap refresh winbox --stable`.
